subprojects/docs/src/docs/userguide/build_cache.adoc

```diff
@@ -191,6 +191,28 @@ We suggest adding tests to ensure that the task inputs are relocatable, i.e. tha
 
 In order to handle volatile inputs for your tasks consider <<more_about_tasks.adoc#sec:configure_input_normalization,configuring input normalization>>.
 
+== Enable caching of non-cacheable tasks
+
+As we have seen, built-in tasks, or tasks provided by plugins, are cacheable if their class is annotated with the `Cacheable` annotation. But what if you want to make cacheable a task whose class is not cacheable? Let's take a concrete example: your build script uses a generic `NpmTask` task to create a JavaScript bundle by delegating to NPM (and running `npm run bundle`). This process is similar to a complex compilation task, but `NpmTask` is too generic to be cacheable by default: it just takes arguments and runs npm with those arguments.
+
+The inputs and outputs of this task are obvious. The input is the directory containing the JavaScript files. The output is the bundle file generated by this task.
+
+.A task that delegates to NPM to bundle the JavaScript files found in a directory
+====
+include::sample[dir="buildCache/non-cacheable-bundle/groovy",files="build.gradle[tags=bundle-task]"]
+include::sample[dir="buildCache/non-cacheable-bundle/kotlin",files="build.gradle.kts[tags=bundle-task]"]
+====
+
+If you're a modern JavaScript developer, you know that bundling can be quite long, and is worth caching. To achieve that, we need to tell Gradle when it's allowed to cache the output of that task, using the link:{javadocPath}/org/gradle/api/tasks/TaskOutputs.html#cacheIf-org.gradle.api.specs.Spec-[TaskOutputs.cacheIf()] method. In this case, it's always allowed. We also need to assign a property name to each input and output, used by the cache to identify them.
+
+This is sufficient to make the task cacheable on your own machine. However, input files are identified by default by their absolute path. So if the cache needs to be shared between several developers or machines using different paths, that won't work as expected. So we also need to set the "path sensitivity". In this case, the relative path of the input files can be used to identify them.
```

Comment:
Let's link to path sensitivity in the build cache guide: https://guides.gradle.org/using-build-cache/#relocatability